translation/verification.py

```diff
@@ -0,0 +1,255 @@
+    return results
+
+
+def translate(
+    client: cohere.client.Client,
+    reference: str,
+    to_lang: str,
+):
+    """Translate text to a target language."""
+    TRANSLATION_PROMPT = Prompt(
+        preamble="""## Instructions
+        You are an expert in translations. Your job is to translate text into a given language.""",
+        message="""{reference}
+        Translate the above message into {to_lang}."""
+    )
+    response = client.chat(
+        chat_history=[
+            {"role": "SYSTEM", "message": TRANSLATION_PROMPT.preamble}
+        ],
+        message=TRANSLATION_PROMPT.message.format(reference=reference, to_lang=to_lang),
+        model=MODEL_NAME
+    )
+    return response
+
+
+def compute_sentence_level_chrf(predictions, references, lowercase=False):
+    scores = []
+    chrf = evaluate.load("chrf")
+    for pred, ref in zip(predictions, references):
+        score = chrf.compute(predictions=[pred], references=[[ref]], word_order=2, lowercase=lowercase)
+        scores.append(score)
+    return scores
+
+
+def check_repeated_tokens(
+    prediction: str, 
+    reference: str,
+    tokenizer: CohereTokenizerFast 
+    ) -> bool:
+    """Check if the prediction has repeated tokens (two or more identical tokens that occur consecutively) that don't appear in the reference text."""
+    prediction_tokens = tokenizer.tokenize(prediction)
+    reference_tokens = tokenizer.tokenize(reference)
+    for i in range(len(prediction_tokens) - 1):
+        if prediction_tokens[i] == prediction_tokens[i + 1] and prediction_tokens[i] not in reference_tokens:
+            return True
+    return False
+
+
+def run_validation(
+    dataset : Dataset,
+    model_id: str = "CohereForAI/aya-23-35B",
+    do_repeated_tokens: bool = True,
+    do_back_translate_chrf: bool = False,
+    do_llm_judge_translation_score: bool = False, 
+    reference_col_name: str = "",
+    prediction_col_name: str  = ""
+) -> dict:
+    """Run validation on a dataset."""
+
+    if do_repeated_tokens:
+        tokenizer = AutoTokenizer.from_pretrained(model_id)
+    
+    co = cohere.Client(api_key=COHERE_API_KEY)
+    if do_llm_judge_translation_score:
+        client = instructor.from_cohere(co)
+
+    references = dataset[reference_col_name]
+    predictions = dataset[prediction_col_name]
+ 
+    repeated_tokens = []
+    back_translate_chrf = []
+    llm_judge_equivalence_of_meaning = []
+    llm_judge_appropriate_wording = []
+    llm_judge_fully_consistent = []
+
+    for reference, prediction in zip(references, predictions):
+        if do_repeated_tokens:
+            repeated = check_repeated_tokens(prediction, reference, tokenizer)
+            repeated_tokens.append(repeated)
+        if do_back_translate_chrf:
+            # for now all backtranslations are into english, add lang id later
+            response = translate(co, prediction, "en")
```

Comment:
```suggestion
            response = translate(co, prediction, "English")
```
Giving language code in the prompt might confuse the model. So, I think spelled out language is better.